reactioncommerce/reaction
# [WIP] feat: promotions framework

Signed-off-by: Brent Hoover <brent@thebuddhalodge.com>

Resolves #6442 #6460 #6461
Impact: **major**
Type: **feature**

## Issue

The PR stakes out the framework for promotions, and for offer triggers which allow you you create rules 

## Solution

This PR uses event listeners to listen for cart update events, evaluates the cart against the active promotions and triggers the emits an event which listeners to handle the actions can listen for.

## Breaking changes

This PR has no breaking changes yet as it is currently all new code.


## Testing
* Edit the plugins.json to add the sample-data plugin, it will add a promotion to the db
* Use the "noop" action (an action for testing)
* Create a cart that satisfies the criteria
* Observe that the noop action was triggered
* Observe that the promotion is stored on the cart
